packages/core-app-api/src/apis/implementations/OAuthRequestApi/MockOAuthApi.ts

```diff
@@ -17,11 +17,19 @@
 import {
   OAuthRequestApi,
   OAuthRequesterOptions,
+  PendingOAuthRequest,
 } from '@backstage/core-plugin-api';
 import { OAuthRequestManager } from './OAuthRequestManager';
 
 export default class MockOAuthApi implements OAuthRequestApi {
   private readonly real = new OAuthRequestManager();
+  private requests: PendingOAuthRequest[] = [];
+
+  constructor() {
+    this.authRequest$().subscribe(requests => {
+      this.requests = requests;
```

Comment:
do we never need to merge with what was there before? as in, pushing the new requests onto the array